### ISSUE TITLE
Add button/link to Help page to join #thelounge

### ISSUE THREAD

**Issue by MaxLeiter** [thelounge/thelounge#2838]
I envision something like this

<img width="575" alt="screen shot 2018-10-09 at 9 57 11 pm" src="https://user-images.githubusercontent.com/8675906/46694948-79362480-cc0e-11e8-802e-673da2df87fd.png">

**Comment by dgw** [comment]
I have some questions about how this is intended to work.

What do you envision this button doing if the user is not connected to freenode?

Assuming the answer is "automatically add the network and connect to it, then join #thelounge": If the user _is_ connected to freenode, but through a non-standard hostname (specific server, IRC bouncer, etc.), what is the user experience then? Duplicate network the user will then have to delete?

Basically, how smart is it going to be about finding an existing connection to freenode and activating or joining #thelounge there?

**Comment by MaxLeiter** [comment]
Yeah thats pretty much what I was planning. Except no need to duplicate network, it would just join #thelounge on the existing Freenode. We could add a little warning explaining what will happened, I.e. if they’re connected to freenode, there’d be a small line above/below the option saying “this channel will be joined on <freenode network name>”

**Comment by dgw** [comment]
So it's looking for any connected network with `NETWORK=freenode` sent in numeric 005? Or just for a connected network using hostname `chat.freenode.net`?

**Comment by MaxLeiter** [comment]
We cant check hostnames because many servers have multiple hosts for different locations/load balancing purposes.

**Comment by damiankw** [comment]
I don't think this is a good idea, from the point of view of someone running TheLounge. Imagine you're running TheLounge for your own network, you wouldn't want someone to be able to use your hosted version to connect to someone elses network. Another point, if TheLounge is running on a host which someone uses as their personal bnc/irc client and a few people happen to connect to freenode it could cause issues with cloning / imitation of the person who hosts it, due to WebIRC not being configured.

Unless the 005/NETWORK = freenode, I think this could bring up a number of different issues. If anything, development should be done on an 'autoconnect' feature (https://thelounge.domain.com/autoconnect) which will connect the client to the server using the default user preferences, and put a link to https://demo.thelounge.chat/ in the About section for a 'Join our channel' link.

**Comment by astorije** [comment]
I can't speak for other @thelounge/maintainers, but I like this idea. The one thing I can think of is this should not be present is `lockNetwork` is enabled and the connected network is not Freenode.

I think having the link say "Join our channel on Freenode" should be enough for people to understand it's on Freenode. We could also have a confirmation modal in case the user is not connected to Freenode already to warn them this will connect to a new server.

> Imagine you're running TheLounge for your own network, you wouldn't want someone to be able to use your hosted version to connect to someone elses network

This is covered by `lockNetwork`. If you don't want users to connect to other networks, this is the option to use.

> If anything, development should be done on an 'autoconnect' feature

Maybe I'm misunderstanding your point, but this seems entirely unrelated to this issue to me, no? That would be #1929.

**Comment by damiankw** [comment]
Regarding the autoconnect, I meant more if lockNetwork was enabled, it could open a new window with demo.thelounge.chat/?nick=<current-nick>&username=<current-username>&autoconnect=true and it would jump straight through to the chat, instead of bringing up a new window for the end user to have to hit connect on, it just means it's more fluent.

Regarding the 'join our channel on Freenode', not entirely true that it should be enough for people to understand it's Freenode. You have to remember that a lot of people using a web chat client don't actually know what IRC is, let alone separate IRC networks. I mean, I don't know the exact user base of this client, but we only use it for our network webchat, end users have no idea. It's locked down, so there's no issues, but I'd say that on a non-locked down client it could be the same, so people might not understand. In saying that, I don't know why non-IRC people would even click the about button to begin with, so maybe your point is valid :P

**Comment by MaxLeiter** [comment]
auto-connect has previously been discussed, and iirc it was denied due to the possibility of abuse.
